control_system_stack/setyaw/src/main.py

```diff
@@ -0,0 +1,185 @@
+        pub_thrusters4.publish(thruster4Data)
+        pub_thrusters6.publish(thruster6Data)
+
+        # Debug messages
+        # rospy.logdebug("--------")
+        # rospy.logdebug("Current Yaw : %s",round(Current_yaw,3))
+        # rospy.logdebug("Desired_yaw : %s",round(self._feedback.Desired_yaw,3))
+        rospy.loginfo("Error : %s",round(YAW.error,3))
+        # rospy.logdebug("Delta_error : %s",round(YAW.delta_error ,3))
+        # rospy.logdebug("Relative_Goal : %s",round(Client_goal.yaw,3))
+        # rospy.logdebug("Thruster data L : %s",thruster6Data.data[4])
+        # rospy.logdebug("Thruster data R : %s",thruster6Data.data[5])
+
+
+
+    def execute_cb(self, goal):
+        """
+        1. Update the goal each time client sends a new goal
+        2. Wait for first imu data to be recieved, to calculate the error.
+        3. calculate the thrust from fuzzy control and send to the thruster converter.
+        4. publishes feedback back to the client
+            - Desired_yaw
+            - Current_yaw
+            - Error
+        """
+
+        global Client_goal
+        global FIRST_ITERATION
+        global Current_yaw
+
+        # Update the goal
+        Client_goal = goal
+        FIRST_ITERATION = True
+        Current_yaw = -1
+
+        rospy.Subscriber(topicHeader.ABSOLUTE_RPY, absoluteRPY, self.imuCB)
+        r = rospy.Rate(20)
+
+        while(Current_yaw == -1):
+            r.sleep()
+
+        rospy.loginfo('I was asked to change yaw by %f', goal.yaw)
+        initial_time = time.time()
+        rospy.loginfo('Started setting yaw at %f', initial_time)
+        success = True
+
+        rospy.loginfo('YAW.error : %s',abs(YAW.error) > 0.5)
+        while abs(YAW.error) > 0.5 :
+            # check that preempt has not been requested by the client
+            # if self._as.is_preempt_requested():
+                # rospy.loginfo('%s: Preempted' % self._action_name)
+                # self._as.set_preempted()
+                # success = False
+                # break
+
+            # publish the feedback
+    		self._as.publish_feedback(self._feedback)
+    		r.sleep()
+
+        self._result.elapsed_time = time.time() - initial_time
+        rospy.loginfo('%s: Succeeded' % self._action_name)
+        self._as.set_succeeded(self._result)
+
+if __name__ == '__main__':
+	"""
+	 1. Declare YAW as a Fuzzy object and Declare it's membership function and it's Range.
+	 2. Declares messages types for thruster4Data and thruster6Data
+	"""
+	YAW = Fuzzy(Fparam.mf_types, Fparam.f_ssets)
+	YAW.io_ranges = Fparam.io_ranges
+
+	thruster4Data=thrusterData4Thruster();
+	thruster6Data=thrusterData6Thruster();
+
+	pub_thrusters4 = rospy.Publisher(topicHeader.CONTROL_PID_THRUSTER4, thrusterData4Thruster, queue_size = 2)
+	pub_thrusters6 = rospy.Publisher(topicHeader.CONTROL_PID_THRUSTER6, thrusterData6Thruster, queue_size = 2)
+
+
+	rospy.init_node('setYaw')
+	SetYaw(rospy.get_name())
```

Comment:
@aniket11bh Write the name of this action server communication explicitly here. The value to be put here is `'setYaw'`. Having a variable such as this is very confusing.